src/query/storages/stage/src/read/row_based/processors/reader.rs

```diff
@@ -48,30 +46,31 @@ impl BytesReader {
     pub fn try_create(
         table_ctx: Arc<dyn TableContext>,
         op: Operator,
-        read_batch_size: usize,
+        _read_batch_size: usize,
         prefetch_num: usize,
     ) -> Result<Self> {
         Ok(Self {
             table_ctx,
             op,
-            read_batch_size,
+            // TODO(xuanwo): bring batch size back when we figure how to play with range based read.
```

Comment:
why not use the setting?

Reply:
Apologies for the confusion; this change is solely for debugging purposes. I have a new implementation locally and will invite you to review it once I've resolved the benchmark issues.